satpy/readers/hdfeos_base.py

```diff
@@ -216,14 +217,48 @@ def load_dataset(self, dataset_name, is_category=False):
         from satpy.readers.hdf4_utils import from_sds
 
         dataset = self._read_dataset_in_file(dataset_name)
-        dask_arr = from_sds(dataset, chunks=CHUNK_SIZE)
+        chunks = self._chunks_for_variable(dataset)
+        dask_arr = from_sds(dataset, chunks=chunks)
         dims = ('y', 'x') if dask_arr.ndim == 2 else None
         data = xr.DataArray(dask_arr, dims=dims,
                             attrs=dataset.attributes())
         data = self._scale_and_mask_data_array(data, is_category=is_category)
 
         return data
 
+    def _chunks_for_variable(self, hdf_dataset):
+        scan_length_250m = 40
+        var_shape = hdf_dataset.info()[2]
+        res_multiplier = self._get_res_multiplier(var_shape)
+        non_yx_chunks = tuple()
+        if len(var_shape) == 3:
+            # assume (band, y, x)
+            non_yx_chunks = ((1,) * var_shape[0],)
+            var_shape = var_shape[-2:]
+        elif len(var_shape) != 2:
```

Comment:
When is len(var_shape) not 2 or 3? For that matter, when is len(var_shape) 2?

Reply:
There are a band number variables:

```
        float Band_250M(Band_250M) ;
                Band_250M:long_name = "250M Band Numbers for Subsetting" ;
        float Band_500M(Band_500M) ;
                Band_500M:long_name = "500M Band Numbers for Subsetting" ;
        float Band_1KM_RefSB(Band_1KM_RefSB) ;
                Band_1KM_RefSB:long_name = "1KM Reflective Solar Band Numbers for Subsetting" ;
        float Band_1KM_Emissive(Band_1KM_Emissive) ;
                Band_1KM_Emissive:long_name = "1KM Emissive Band Numbers for Subsetting" ;
```

but we don't currently use them as far as I know. Overall I just didn't like leaving a case that wasn't handled and rather than fail loudly I thought I'd just do auto chunking.